deluge/ui/common.py

```diff
@@ -264,6 +264,22 @@ def walk(path, item):
                 file_tree = FileTree(paths)
                 file_tree.walk(walk)
             self._files_tree = file_tree.get_tree()
+        elif 'file tree' in info_dict:
+            if len(info_dict['file tree']) > 1:
+                info_dict['file tree'] = {info_dict['name']: info_dict['file tree']}
```

Comment:
This doesn't work for a rootless multifile torrent with a single file. https://dev.deluge-torrent.org/ticket/3602

From just parsing the 'file tree' key I don't think we can get the relevant info. This might be a better check

```suggestion
            def single_file_torrent(info_dict):
                if len(info_dict['file tree']) > 1:
                    return False
                
                filename = [key for key in info_dict['file tree']][0]
                return info_dict['name'] == filename

            if not single_file_torrent(info_dict):
                info_dict['file tree'] = {info_dict['name']: info_dict['file tree']}
```

I'm thinking that single_file_torrent should be a static method in FileTree v2 class